sample_caller.py

```diff
@@ -24,12 +65,72 @@ def signal_handler(signal, frame):
     die_now = True
 
 
-def extract_doc_content(data):
-    # Your data extraction routines here. (pdftotext, abiword, etc.)
-    return data
+def print_dict(dic: dict):
+    for k, v in dic.items():
+        if isinstance(v, str):
+            value = trunc(v, 200, ellipsis="...")
+            v_print(1, f'    {k}: "{value}"')
+        else:
+            # Dates and such...
+            v_print(1, f"    {k}: {v}")
 
 
-def scrape_court(site, binaries=False):
+def extract_doc_content(
+    data, extract_from_text: bool, site, doctor_host: str, filename: str
+):
+    """Extracts document's content using a local doctor host
+
+    For complete and integrated testing, run in a Courtlistener
+    docker compose environment
+
+    :param data: the response content
+    :param extract_from_text: if True, extract doc content using doctor
+        if False, return content as is
+    :param site: current site object
+    :param doctor_host: local doctor instance host. calls will fail if
+        the doctor host is not valid
+    :param filename: Name for saving extracted content into a file in tmp
+
+    :return: a tuple with:
+        the extracted content
+        the structured metadata parsed by Site.extract_from_text
+    """
+    if not extract_from_text:
+        return data, {}
+
+    # Get the file type from the document's raw content
+    extension_url = MICROSERVICE_URLS["buffer-extension"]["url"].format(
+        doctor_host
+    )
+    extension_response = requests.post(
+        extension_url, files={"file": ("filename", data)}, timeout=30
+    )
+    extension_response.raise_for_status()
+    extension = extension_response.text
+
+    # For HTML files, the extracted document may have modified tags
+    files = {"file": (f"something.{extension}", data)}
+    url = MICROSERVICE_URLS["document-extract"]["url"].format(doctor_host)
+    extraction__response = requests.post(url, files=files, timeout=120)
+    extraction__response.raise_for_status()
+    extracted_content = extraction__response.json()["content"]
+
+    # The extracted content is embeded for display in Courtlistener
+    # We save it into /tmp/ to have an idea how it would look. You can
+    # inspect it your browser by going into f'file://tmp/{filename}.html'
+    court_id = site.court_id.split(".")[-1]
+    with open(f"/tmp/{court_id}_{filename}.html", "w") as f:
```

Comment:
You are going to need to modify the filename - I tested this with Arkansas and got a filename too long error - cause the case name was novel length.